### ISSUE TITLE
construct-constructor-property / no-unused-props: constructor overload signature shadows the implementation

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#533]
### Describe the bug

`findConstructor` (`packages/*/src/core/ast-node/finder/constructor.ts`) returns the **first** class member whose `kind` is `"constructor"`. When a constructor has overload signatures, the first member is the overload *signature* (with no body), not the implementation. As a result:

- `construct-constructor-property` validates only the first overload signature and never checks the implementation signature (false negative)
- `no-unused-props` analyzes a constructor with `body: null`, so no usage is ever recorded and every property is reported as unused (false positive)

**Repro 1 — construct-constructor-property (false negative):**

```ts
import { Construct } from "constructs";

export class OverloadCtor extends Construct {
  constructor(scope: Construct, id: string);
  constructor(parent: Construct, name: string, opts?: number) {
    super(parent, name);
    void opts;
  }
}
```

**Repro 2 — no-unused-props (false positive):**

```ts
import { Bucket } from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";

interface OverloadProps {
  readonly bucketName: string;
}

export class OverloadConstruct extends Construct {
  constructor(scope: Construct, id: string, props: OverloadProps);
  constructor(scope: Construct, id: string, props: OverloadProps, extra?: number) {
    super(scope, id);
    new Bucket(this, "Dest", { bucketName: props.bucketName });
    void extra;
  }
}
```

### Expected Behavior

- Repro 1: the implementation parameters `parent` / `name` / `opts` violate the `scope, id, props` convention and should be reported.
- Repro 2: no error — `bucketName` is used in the implementation body.

### Current Behavior

- Repro 1: no error at all (both plugins).
- Repro 2:

```
repro/no-unused-props/overload.ts  10:45  error  Property 'bucketName' is defined in props but never used  awscdk/no-unused-props
```

Oxlint reports the same (`repro/no-unused-props/overload.ts:10:45`).

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a), macOS, Node.js 24.18.0, eslint 10.8.1, oxlint 1.79, rule-isolated configs.
- Root cause: `packages/eslint/src/core/ast-node/finder/constructor.ts` (`.find()` on `kind === "constructor"`); identical in `packages/oxlint`. Overload signatures are `MethodDefinition`s whose value is a `TSEmptyBodyFunctionExpression`, so the finder should skip members without a body.
- `props-name-convention` uses the same finder and is affected the same way (name check runs against the overload signature instead of the implementation).
